web_src/js/features/comp/ComboMarkdownEditor.js

```diff
@@ -246,6 +277,24 @@ class ComboMarkdownEditor {
     }
   }
 
+  textareaAutoResize() {
+    if (this.textareaInitalHeight === undefined) {
+      this.textareaInitalHeight = this.textarea.offsetHeight;
+    }
+    const offset = this.textarea.offsetHeight - this.textarea.clientHeight;
+    if (!this.lastValue || Math.abs(this.lastValue.length - this.textarea.value.length) > 2) {
+      // the value has changed a lot, so reset the height to calculate the real scroll height, it might cause UI flickering
+      this.textarea.style.height = 'auto';
+    } else {
+      // try to shrink a little to see if a line is deleted (since the value doesn't change much), it won't cause UI flickering
+      // the magic number is a general number which fits most line-height styles.
+      this.textarea.style.height = `${this.textarea.scrollHeight + offset - 40}px`;
+    }
+    // make sure the height is not smaller than the initial height
+    this.textarea.style.height = `${Math.max(this.textareaInitalHeight, this.textarea.scrollHeight + offset)}px`;
+    this.lastValue = this.textarea.value;
+  }
```

Comment:
Maybe outsource this to https://github.com/jackmoore/autosize?

Reply:
Yeah I guess if we can do without, it'll be better. I recall I've implemented something similar on another project. On `input` event, trigger this code. The values depend on styling of the element so will be different for us.

```js
el.style.height = "auto";
const maxHeight = 17.6/* padding */ + 16 * 1.4 * maxRows/* row with line-height */;
const height = el.scrollHeight;
el.style.height = `${Math.min(height, maxHeight)}px`;
el.style.maxHeight = `${maxHeight}px`;
if (height <= maxHeight) {
  el.style.overflowY = "hidden";
} else {
  el.style.removeProperty("overflow-y");
}
```